File: src/#559/solution.js
function Node(val, children) {
  this.val = val
  this.children = children
}

var maxDepth = function (root) {
  if (root === null) {
    return 0
  }
  var m = 1
  for (let i = 0; i < root.children.length; i++) {
    m = Math.max(m, maxDepth(root.children[i]) + 1)
  }
  return m
}

File: src/#559/solution.go
package solution

type Node struct {
	Val      int
	Children []*Node
}

func maxDepth(root *Node) int {
	if root == nil {
		return 0
	}
	m := 1
	for _, n := range root.Children {
		m = max(m, maxDepth(n)+1)
	}
	return m
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
